textpipe/doc.py

```diff
@@ -132,13 +154,29 @@ def clean_text(self, remove_html=True, clean_dots=True, clean_quotes=True,
     @property
     def ents(self):
         """
-        Extract a list of the named entities in text
+        A list of the named entities with sensible defaults.
 
         >>> doc = Doc('Sentence for testing Google text')
         >>> doc.ents
         [('Google', 'ORG')]
         """
-        return list({(ent.text, ent.label_) for ent in self.spacy_doc.ents})
+        return self.find_ents()
+
+    @hash_dict
+    @functools.lru_cache()
+    def find_ents(self, model_mapping=None):
+        """
+        Extract a list of the named entities in text, with the possibility of using a custom model.
+        >>> doc = Doc('Sentence for testing Google text')
+        >>> doc.find_ents()
+        [('Google', 'ORG')]
+        """
+        lang = self.hint_language if self.language == 'un' else self.language
+        model_name = model_mapping.get(lang, None) if model_mapping else None
+        try:
+            return list({(ent.text, ent.label_) for ent in self.spacy_doc(model_name).ents})
+        except TextpipeMissingModelException:
+            return list({(ent.text, ent.label_) for ent in self.spacy_doc().ents})
```

Comment:
This would silently fall back on a SpaCy model. Not sure if this is wanted behavior. For example if you have a solely financial entity model, you'd be getting noisy date from this.

Reply:
You're right; I won't catch this exception here.